lib/topologies/shared.js

```diff
@@ -56,6 +56,20 @@ function createClientInfo(options) {
   return clientInfo;
 }
 
+function createCompressionInfo(options) {
+  if ("compression" in options && "compressors" in options.compression) {
```

Comment:
I don't recommend using in as it can give false positives https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in It might be better to just use indexOf to avoid the side effects of in.